app/views/application/_breadcrumbs_and_title.html.slim

```diff
@@ -1 +1,50 @@
-- content_for :title, application_name
+- model = controller_path.classify.constantize rescue nil
+- if model
+  - object = model.find_by(id: params[:id])
+  - if (parent_model = model.try(:parent_resource))
+    - parent_route_key = parent_model.model_name.singular_route_key
+    - if params["#{parent_route_key}_id"].present?
+      - parent_object = object.try(parent_route_key) || parent_model.find_by(id: params["#{parent_route_key}_id"])
+      - parent_element = link_to_if(current_user && can?(:show, parent_object), parent_object, {controller: parent_model.model_name.route_key, action: :show, id: parent_object.id})
+      - parent_root_element = link_to_if(current_user && can?(:index, parent_model), parent_model.model_name.human(count: 2), {controller: parent_model.model_name.route_key, action: :index})
+      - root_element = link_to_if(current_user && can?(:index, model), model.model_name.human(count: 2), send(:"#{parent_route_key}_#{model.model_name.route_key}_path", parent_object))
+      - if object
+        - current_element = link_to_if(current_user && can?(:show, object), object, send(:"#{parent_route_key}_#{model.model_name.singular}_path", parent_object, object))
+    - else
+      - root_element = link_to_if(current_user && can?(:index, model), model.model_name.human(count: 2), {controller: model.model_name.route_key, action: :index})
+      - if object
+        - current_element = link_to_if(current_user && can?(:show, object), object, {controller: model.model_name.route_key, action: :show, id: object.id})
+  - else
+    - root_element = link_to_if(current_user && can?(:index, model), model.model_name.human(count: 2), {controller: model.model_name.route_key, action: :index})
+    - if object
+      - current_element = link_to_if(current_user && can?(:show, object), object, {controller: model.model_name.route_key, action: :show, id: object.id})
+  - if I18n.exists?("shared.#{params[:action]}")
+    - active_action = t("shared.#{params[:action]}", model: model&.model_name&.human)
+  - else
+    - active_action = t("#{controller_name}.index.#{params[:action]}", model: model&.model_name&.human)
+- else
+  - active_action = t("breadcrumbs.#{controller_name}.#{params[:action]}")
+
+
+- title = "#{active_action} - #{application_name}"
```

Comment:
Personally I think "Tasks - CodeOcean" would read better than "Tasks - Index - CodeOcean". Similarly we might omit the show action because in that case we have the resource name (for example "Example Task - CodeOcean" instead of "Example Task - Show - CodeOcean"). That is just preference though.

Reply:
That's a valid suggestion, which I implemented with 02c1bc4d.